pkg/tenant/db/db.go

```diff
@@ -301,28 +306,37 @@ func deleteRequest(ctx context.Context, namespace string, key []byte) (err error
 	return nil
 }
 
-func List(ctx context.Context, prefix []byte, namespace string) (values [][]byte, err error) {
+func List(ctx context.Context, prefix, seekKey []byte, namespace string, onListItem OnListItem, c *pg.Cursor) (cursor *pg.Cursor, err error) {
 	mu.RLock()
 	defer mu.RUnlock()
 
 	if !connected() {
 		return nil, ErrNotConnected
 	}
 
+	// Check to see if a default cursor exists and create one if it does not.
+	if c == nil {
+		c = pg.New("", "", 0)
+	}
+
+	if c.PageSize <= 0 {
+		return nil, ErrMissingPageSize
```

Comment:
Instead of erroring, I recommend simply setting the page size to a default page size.